Makefile

```diff
@@ -3,6 +3,28 @@
 # This source code is licensed under the Apache License, Version 2.0 found in
 # the LICENSE.txt file in the root directory of this source tree.
 
+OS := $(shell uname -s)
+
+ifeq ($(OS), Linux)
+  NPROCS := $(shell grep -c ^processor /proc/cpuinfo)
+else ifeq ($(OS), Darwin)
+  NPROCS := 2
+else
+  NPROCS := 0
+endif # $(OS)
+
+ifeq ($(NPROCS), 2)
+	CONCURRENCY := 2
+else ifeq ($(NPROCS), 1)
+	CONCURRENCY := 1
+else ifeq ($(NPROCS), 3)
+	CONCURRENCY := 3
+else ifeq ($(NPROCS), 0)
+	CONCURRENCY := 0
+else
+	CONCURRENCY := $(shell echo "$(NPROCS) 2" | awk '{print $$1 / $$2}')
```

Comment:
We should make sure to get an integer back from awk here.

Reply:
Done using printf in: https://github.com/Qiskit/qiskit-terra/pull/1973/commits/12918d5cefd51356ead703c59af10128c7ab5b97